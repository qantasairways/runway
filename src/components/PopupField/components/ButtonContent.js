import React from 'react';
import PropTypes from 'prop-types';

import ButtonValues from './ButtonValues';

import {
  fontSize,
  colours,
  buttons,
  letterSpacing
} from '../../../theme/airways';

function ButtonContent({ fieldLabel, placeHolder, values, Icon }) {
  const [firstValue, secondValue] = values;

  return (
    <div
      css={{
        height: buttons.heightLarge,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        maxWidth: '100%'
      }}
    >
      <span
        css={{
          label: 'runway-popup-field__label',
          color: 'white',
          textTransform: 'uppercase',
          position: 'absolute',
          top: '5px',
          left: '7px',
          fontSize: fontSize.label
        }}
      >
        {fieldLabel}
      </span>
      {firstValue && (
        <ButtonValues firstValue={firstValue} secondValue={secondValue} />
      )}
      {!firstValue && (
        <div
          css={{
            label: 'runway-popup-field__placeholder',
            color: colours.darkGrey,
            fontSize: fontSize.large,
            letterSpacing: letterSpacing.small
          }}
        >
          {placeHolder}
        </div>
      )}
      {!firstValue && !!Icon && (
        <Icon
          css={{
            label: 'runway-popup-field__icon',
            position: 'absolute',
            right: '10px',
            fill: colours.darkGrey
          }}
          height="36"
          width="36"
        />
      )}
    </div>
  );
}

ButtonContent.propTypes = {
  fieldLabel: PropTypes.string,
  placeHolder: PropTypes.string,
  values: PropTypes.arrayOf(
    PropTypes.shape({
      large: PropTypes.string,
      small: PropTypes.string
    })
  ),
  Icon: PropTypes.func
};

ButtonContent.defaultProps = {
  fieldLabel: null,
  placeHolder: null,
  values: [],
  Icon: null
};

export default ButtonContent;
