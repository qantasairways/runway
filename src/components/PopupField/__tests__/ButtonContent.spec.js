import React from 'react';
import { mount } from 'enzyme';
import ButtonContent from '../components/ButtonContent';

const DummyIcon = () => <div>ICON</div>;

describe('ButtonContent', () => {
  let component;

  it('renders correctly with defaults', () => {
    component = mount(<ButtonContent />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with props provided', () => {
    const fieldLabel = 'Departing';
    const placeHolder = 'Departing From?';

    component = mount(
      <ButtonContent
        fieldLabel={fieldLabel}
        placeHolder={placeHolder}
        Icon={DummyIcon}
      />
    );

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with values provided', () => {
    const fieldLabel = 'Departing';
    const placeHolder = 'Departing From?';
    const values = [
      { large: '18', small: 'Tue, May 2019' },
      { large: '20', small: 'Thu, May 2019' }
    ];

    component = mount(
      <ButtonContent
        fieldLabel={fieldLabel}
        placeHolder={placeHolder}
        values={values}
        Icon={DummyIcon}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
