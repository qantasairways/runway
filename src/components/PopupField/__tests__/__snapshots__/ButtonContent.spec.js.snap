// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContent renders correctly with defaults 1`] = `
.emotion-2 {
  height: 6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 100%;
}

.emotion-0 {
  color: white;
  text-transform: uppercase;
  position: absolute;
  top: 5px;
  left: 7px;
  font-size: 1rem;
}

.emotion-1 {
  color: #626262;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
}

<ButtonContent
  Icon={null}
  fieldLabel={null}
  placeHolder={null}
  values={Array []}
>
  <div
    className="emotion-2"
  >
    <span
      className="emotion-0"
    />
    <div
      className="emotion-1"
    />
  </div>
</ButtonContent>
`;

exports[`ButtonContent renders correctly with props provided 1`] = `
.emotion-3 {
  height: 6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 100%;
}

.emotion-0 {
  color: white;
  text-transform: uppercase;
  position: absolute;
  top: 5px;
  left: 7px;
  font-size: 1rem;
}

.emotion-1 {
  color: #626262;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
}

.emotion-2 {
  position: absolute;
  right: 10px;
  fill: #626262;
}

<ButtonContent
  Icon={[Function]}
  fieldLabel="Departing"
  placeHolder="Departing From?"
  values={Array []}
>
  <div
    className="emotion-3"
  >
    <span
      className="emotion-0"
    >
      Departing
    </span>
    <div
      className="emotion-1"
    >
      Departing From?
    </div>
    <DummyIcon
      className="emotion-2"
      height="36"
      width="36"
    >
      <div>
        ICON
      </div>
    </DummyIcon>
  </div>
</ButtonContent>
`;

exports[`ButtonContent renders correctly with values provided 1`] = `
.emotion-9 {
  height: 6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  max-width: 100%;
}

.emotion-0 {
  color: white;
  text-transform: uppercase;
  position: absolute;
  top: 5px;
  left: 7px;
  font-size: 1rem;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  overflow: hidden;
  max-width: 100%;
}

.emotion-4 {
  color: white;
  margin: 0 15px;
  font-size: 36px;
  margin-bottom: 20px;
}

.emotion-3 {
  max-width: 100%;
  min-width: 0;
}

.emotion-1 {
  color: #fff;
  font-size: 2.25rem;
  -webkit-letter-spacing: -0.0625rem;
  -moz-letter-spacing: -0.0625rem;
  -ms-letter-spacing: -0.0625rem;
  letter-spacing: -0.0625rem;
  line-height: 2.65rem;
}

.emotion-2 {
  color: #fff;
  font-size: 1.125rem;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  max-width: 100%;
}

<ButtonContent
  Icon={[Function]}
  fieldLabel="Departing"
  placeHolder="Departing From?"
  values={
    Array [
      Object {
        "large": "18",
        "small": "Tue, May 2019",
      },
      Object {
        "large": "20",
        "small": "Thu, May 2019",
      },
    ]
  }
>
  <div
    className="emotion-9"
  >
    <span
      className="emotion-0"
    >
      Departing
    </span>
    <ButtonValues
      firstValue={
        Object {
          "large": "18",
          "small": "Tue, May 2019",
        }
      }
      secondValue={
        Object {
          "large": "20",
          "small": "Thu, May 2019",
        }
      }
    >
      <div
        className="emotion-8"
      >
        <ButtonValue
          large="18"
          small="Tue, May 2019"
        >
          <div
            className="emotion-3"
          >
            <div
              className="emotion-1"
            >
              18
            </div>
            <div
              className="emotion-2"
            >
              Tue, May 2019
            </div>
          </div>
        </ButtonValue>
        <div
          className="emotion-4"
        >
          -
        </div>
        <ButtonValue
          large="20"
          small="Thu, May 2019"
        >
          <div
            className="emotion-3"
          >
            <div
              className="emotion-1"
            >
              20
            </div>
            <div
              className="emotion-2"
            >
              Thu, May 2019
            </div>
          </div>
        </ButtonValue>
      </div>
    </ButtonValues>
  </div>
</ButtonContent>
`;
